src/StackExchange.Redis/APITypes/BitfieldSubCommand.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace StackExchange.Redis;
+
+/// <summary>
+/// An abstract subcommand for a bitfield.
+/// </summary>
+public abstract class BitfieldSubCommand
+{
+    internal abstract int NumArgs { get; }
+
+    internal abstract void AddArgs(IList<RedisValue> args);
+
+    internal virtual bool IsReadonly => false;
+
+    /// <summary>
+    /// The encoding of the sub-command. A signed or unsigned integer of a given size.
+    /// </summary>
+    public BitfieldEncoding Encoding { get; }
+
+    /// <summary>
+    /// The offset into the bitfield the subcommand will traverse.
+    /// </summary>
+    public BitfieldOffset Offset { get; }
+
+    internal BitfieldSubCommand(BitfieldEncoding encoding, BitfieldOffset offset)
+    {
+        Encoding = encoding;
+        Offset = offset;
+    }
+
+}
+
+/// <summary>
+/// Represents a Bitfield GET, which returns the number stored in the specified offset of a bitfield at the given encoding.
+/// </summary>
+public sealed class BitfieldGet : BitfieldSubCommand
```

Comment:
I think the approach is a bit heavy here - we're doing a lot of class creation on what seems like should be structs local to the command generation itself and not public API surface area. I'm happy to yank this local and try other approaches - will try and play some tonight.

Reply:
Interested to see what you come up with, this seemed liked best way to me. It's a weird command because of how variadic and structured it is. There's other commands where you just pass in arrays of literals and you can work everything out from that, but in this case you're really expecting things in a very specific order. Kind of tricky to design an API around, might be why it's been around for 6 years without making it into the library. I will tell you though I give a talk on bit-operations and the inability to use bitfield natively has been a real bummer 😆 so I have a bit of skin in this one.

Reply:
My thinking is: these are Get*Message methods in RedisDatabase like others - they are just specific overloads. I don't know why we'd need classes on these (maybe should change the implementation on the other PR already in as well). This approach allocates another object to make the message as well as exposes them on the public API but as far as I can tell they don't need to be public. I think we can simplify these subcommands to some of those methods and remove the extra classes/allocations here overall.

Reply:
Maybe for the individual commands yes, but the command itself is weirdly variadic (you can execute an arbitrary number of subcommand, it's almost like a script), so you need some structure to maintain them all in, which is the thinking behind having them all broken out into different classes with an abstract driver. The alternative would be to just expose the single-commands and just let folks batch/pipeline them, but it sort of breaks the command's API.

Reply:
Well I poked at this for a bit and then git just royally screwed me with a UI change recently on main vs. current branch merging in my tool and lost everything so I'm hands off tonight. Basically I had BitfieldSubCommand -> IBitfieldSubCommand, internal members on that interface still, explicitly implemented on members, and all the API changes that entails. However, I'm not sure this is a good path or not - I don't like the amount of allocations we're doing for what is intended to be, on the Redis side, a very efficient/optimized op. Generally our audiences who want this functionality will also care about the cost (I know I would).

Overall, this one may wait, would like to talk it over with @mgravell on surface area and I'm supposed to be out this week. Given git giving me signs, gonna step away for now :)